### DISCUSSION TITLE
Replace use statement by fully qualified name

### DISCUSSION THREAD

**Discussion by dimitriBouteille** [PHP-CS-Fixer/PHP-CS-Fixer#6813] [Q&A]
Hey, Is there a rule to replace use with fully qualified name ?

**input :**  

```php
<?php

namespace Test;

use Magento\Store\Model\Store;

class Test
{

    public function foo()
    {
        return new Store();
    }
}
```

**output:**

```php
<?php

namespace Test;

class Test
{

    public function foo()
    {
        return new \Magento\Store\Model\Store();
    }
}
```

**Comment by kubawerlos** [comment]
Hi @dimitriBouteille,

there is no such rule currently. Why do you want to do this, is it a part of a standard of some community?

**Comment by dimitriBouteille** [comment]
Hello @kubawerlos 

On some project (like projects done on Magento) there are several times the same class name that can be imported.

In such cases, it is easier and more convenient to use the full name of the class.

Here's an example:

```php

namespace Reflet\AdminGws;

class Test
{
    protected \Reflet\Faq\Model\Faq $model;
    protected \Reflet\Faq\Model\ResourceModel\Faq $resource;
}
```

With imports it’s a little more complicated to understand the code

```php
namespace Reflet\AdminGws;

use \Reflet\Faq\Model\Faq as Model;
use \Reflet\Faq\Model\ResourceModel\Faq as Resource;

class Test
{
    protected Model $model;
    protected Resource $resource;
}
```

**Reply by Wirone** [reply]
Import aliasing is available exactly to solve this naming conflicts. There's no point IMHO in using FQCN everywhere just to avoid name conflicts. We don't have such rule and I don't think anyone from the core team would want to work on it. In fact, I introduced `import_symbols` option to `fully_qualified_strict_types` fixer to do exactly opposite 😊.
